services/github-service-api/src/main/java/org/kontinuity/catapult/service/github/api/GitHubRepository.java

```diff
@@ -23,4 +24,18 @@
      * @return the github.com page for the repository
      */
     URI getHomepage();
+
+    /**
+	 * @param path
+	 *            the path to the file
+	 * @return the URL of the file with the given {@code} in the GitHub
+	 *         repository, <code>null</code> if no file with the given
+	 *         {@code path} exists in the repository.
+	 */
+	URI getDownloadUrl(String path);
```

Comment:
url/uri 

And do we want to throw an exception here if requesting a download URI for a code path that doesn't exist?  Returning nulls encourages NPE.